### ISSUE TITLE
"A[:] = ..." allocates memory

### ISSUE THREAD

**Issue by stevengj** [JuliaLang/julia#16302]
It seems like this construct should operate in-place, but it apparently doesn't:
```jl
function foo(A)
    for i = 1:10^6
        A[:] = 0
    end
end
@time foo([3,4,5])
```
gives
```
0.021217 seconds (1.00 M allocations: 15.259 MB, 23.12% gc time)
```
Notice that it is apparently allocating something on each loop iteration.  (Same on 0.4 and 0.5.)

**Comment by yuyichao** [comment]
The allocation is caused by the vararg call.

**Comment by stevengj** [comment]
We should have a specialized `setindex` method for this case that just calls `fill!`.

**Comment by yuyichao** [comment]
Ref https://github.com/JuliaLang/julia/pull/13459#issuecomment-145839032 , maybe .... It looks like we are getting close to fix #13359 though....

**Comment by stevengj** [comment]
Closing in favor of #13459.
